### PR TITLE
Features/npm update dependence

### PR DESCRIPTION
**Pull request by sanjayradadiya**
Fixes #

Changes in this pull request:
- Major updates (requires more testing and changes)
- Removed potentially unused modules (NOTUSED?)
- Added used, but only implicitly loaded modules

@norbertschuler @harshvitra @ankitecd

### CODE REVIEW

**Comment by sanjayradadiya** [comment]
https://www.loom.com/share/b13f3369aeba4e3ba695f81af6fca304

**Comment by harshvitra** [comment]
Screenshot_2020-03-07-02-11-36-879_org pftp

I'm immediately getting this error

File: package.json [hunk 1]
```diff
@@ -95,77 +94,77 @@
     "react-native-raw-bottom-sheet": "^2.0.6",
     "react-native-reanimated": "^1.7.0",
     "react-native-root-toast": "^3.2.0",
-    "react-native-screens": "^1.0.0-alpha.23",
-    "react-native-simple-radio-button": "^2.7.4",
+    "react-native-safe-area-context": "^0.7.3",
```
Comment (norbertschuler) [inline package.json]: I can not find `react-native-safe-area-context` to be used?

File: package.json [hunk 2]
```diff
@@ -95,77 +94,77 @@
...
     "redux-actions": "^2.6.5",
     "redux-form": "^8.3.0",
     "redux-logger": "^3.0.6",
     "redux-thunk": "^2.3.0",
     "reselect": "^4.0.0",
     "rn-fetch-blob": "^0.12.0",
-    "rxjs": "^5.5.12",
-    "scheduler": "^0.18.0",
+    "rxjs": "^6.5.4",
+    "rxjs-compat": "^6.5.4",
```
Comment (norbertschuler) [inline package.json]: I cannot find a usage of `rxjs-compat`?

File: app/helpers/reduxHelpers.native.js [hunk 3]
```diff
@@ -1,13 +1,12 @@
...
+  createReduxContainer
 } from 'react-navigation-redux-helpers';
 
 const drawerNavMiddleware = createReactNavigationReduxMiddleware(
-  'drawerRoot',
   state => state.appDrawer
 );
 
-const drawerRootListener = createReduxBoundAddListener('drawerRoot');
+const drawerRootListener = createReduxContainer('drawerRoot');
```
Comment (norbertschuler) [inline app/helpers/reduxHelpers.native.js]: I am not completely sure, but I think the exported object `drawerRootListener` is never used as the only usage of this code I found is in https://github.com/Plant-for-the-Planet-org/treecounter-app/blob/develop/app/stores/middlewares.native.js.

**Comment by norbertschuler** [comment]
Got an error in the web client on the register screen - probably from the npm package `uuidv4`:
```
Uncaught TypeError: (0 , _uuidv.default) is not a function
    at SignUp.componentDidMount (SignUp.js:30)
    at commitLifeCycles (react-dom.development.js:22101)
    at commitLayoutEffects (react-dom.development.js:25344)
    at HTMLUnknownElement.callCallback (react-dom.development.js:336)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:385)
    at invokeGuardedCallback (react-dom.development.js:440)
    at commitRootImpl (react-dom.development.js:25082)
    at unstable_runWithPriority (scheduler.development.js:697)
    at runWithPriority$2 (react-dom.development.js:12149)
    at commitRoot (react-dom.development.js:24922)
```

**Comment by norbertschuler** [comment]
> Got an error in the web client on the **register** and the **login** screen - probably from the npm package `uuidv4`:

Fixed.

File: app/helpers/reduxHelpers.native.js (hunk 3 above)
Comment (norbertschuler) [reply app/helpers/reduxHelpers.native.js]: As this is causing the above error mentioned in https://github.com/Plant-for-the-Planet-org/treecounter-app/pull/2016#issuecomment-595956979 I would just delete this code for now. Or if someone could study the documentation of `redux-helpers`, change the code the way it really should be done :-)

File: package.json [hunk 4]
```diff
@@ -29,64 +29,63 @@
...
     "@esri/react-arcgis": "^5.0.0",
-    "@react-native-community/async-storage": "^1.8.0",
-    "@react-native-community/cli-platform-android": "^3.1.4",
-    "@react-native-community/cli-platform-ios": "^3.2.0",
+    "@react-native-community/async-storage": "^1.8.1",
+    "@react-native-community/cli-platform-android": "^4.2.2",
+    "@react-native-community/cli-platform-ios": "^4.2.2",
+    "@react-native-community/datetimepicker": "^2.2.2",
     "@react-native-community/geolocation": "^2.0.2",
+    "@react-native-community/masked-view": "^0.1.6",
```
[the remark this answers is not in the document]
Comment (sanjayradadiya) [reply package.json]: it's required by https://www.npmjs.com/package/react-navigation-stack

File: package.json (hunk 1 above)
Comment (sanjayradadiya) [reply package.json]: it's required by https://www.npmjs.com/package/react-navigation-stack

**Comment by norbertschuler** [comment]
I have an issue on Android API 28 with my emulator as following:
```
[Error: crypto.getRandomValues() not supported. See https://github.com/uuidjs/uuid#getrandomvalues-not-supported]
```

**Comment by norbertschuler** [comment]
> [Error: crypto.getRandomValues() not supported. See https://github.com/uuidjs/uuid#getrandomvalues-not-supported]

Fixed it as describe in https://github.com/uuidjs/uuid#getrandomvalues-not-supported. Maybe we don't need to import `uuidv4` as `uuid` also has v4 support like `import { v4 as uuidv4 } from 'uuid';`

File: app/helpers/reduxHelpers.native.js (hunk 3 above)
Comment (norbertschuler) [reply app/helpers/reduxHelpers.native.js]: I did not get the error as seen by @harshvitra when starting the iOS or Android app.

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (norbertschuler) [reply package.json]: Isn’t it automatically loaded by that module?

[the remark this answers is not in the document]
Comment (norbertschuler) [reply package.json]: O.k., I read https://github.com/react-navigation/stack

**Comment by norbertschuler** [comment]
> On app load, I'm immediately getting this error
> Tried to reload but this stays

@harshvitra do you still get an error? Otherwise I would merge this PR now despite @sanjayradadiya has not yet commented on all of my review comments.

File: app/helpers/reduxHelpers.native.js (hunk 3 above)
Comment (norbertschuler) [reply app/helpers/reduxHelpers.native.js]: I removed the unnecessary usage of createReduxContainer now.

File: package.json (hunk 2 above)
Comment (norbertschuler) [reply package.json]: I ignore this for now (as the package is not that big)

**Pull request merged**